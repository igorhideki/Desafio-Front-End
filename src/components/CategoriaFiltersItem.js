import React from 'react';
import PropTypes from 'prop-types';

const handleIsActive = (isActive, componentValue) => {
  return isActive === componentValue ? 'filter__item--active' : '';
};

const CategoriaFiltersItem = (props) => (
  <span className={`filter__item ${handleIsActive(props.filterActive, props.type)}`} onClick={() => {
    props.handleSelectCategoria(props.type);
  }}>
    {props.name}
  </span>
);

CategoriaFiltersItem.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.number.isRequired,
  filterActive: PropTypes.number,
  handleSelectCategoria: PropTypes.func  
};


export default CategoriaFiltersItem;
