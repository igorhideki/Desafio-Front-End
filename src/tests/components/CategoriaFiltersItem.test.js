import React from 'react';
import { shallow } from 'enzyme';
import CategoriaFiltersItem from '../../components/CategoriaFiltersItem';
import categorias from '../fixtures/categorias';

test('should render CategoriaFiltersItem correctly', () => {
  const wrapper = shallow(<CategoriaFiltersItem{...categorias[1]}/>);
  expect(wrapper).toMatchSnapshot();
});

