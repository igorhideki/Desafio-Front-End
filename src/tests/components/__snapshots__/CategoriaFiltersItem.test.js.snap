// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CategoriaFiltersItem correctly 1`] = `
<span
  className="filter__item "
  onClick={[Function]}
>
  For√ßa de Vendas
</span>
`;
